src/providers/ldap/sdap_async.c

```diff
@@ -80,6 +80,16 @@ static void sdap_call_op_callback(struct sdap_op *op, struct sdap_msg *reply,
               "Handling LDAP operation [%d][%s] took [%"PRIu64"] microseconds.\n",
               op->msgid, op->stat_info == NULL ? "-" : op->stat_info,
               time_spend);
+
+        /* time_spend is in us and timeout in s */
+        if (op->timeout != 0 && (time_spend / op->timeout) >= (80 * 10000)) {
+            DEBUG(SSSDBG_IMPORTANT_INFO, "LDAP operation [%d][%s] seems slow, "
+                                         "took more than 80%% of timeout.\n",
+                                         op->msgid,
+                                         op->stat_info == NULL ? "-"
+                                                               : op->stat_info);
```

Comment:
Having additional local `const char *info = ...` would simplify reading code a bit, imo.

Reply:
added

Reply:
I meant `const char *info = (op->stat_info == NULL ? "-" : op->stat_info)`; to avoid copy&paste of `...== NULL ? "-"...`...